internal-services/catalog/create-advisory-task.yaml

```diff
@@ -0,0 +1,115 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: task-create-advisory
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+      Pushes an advisory yaml to a Git repository
+  params:
+    - name: repo
+      type: string
+      description: Git repository to push the advisory to
+    - name: advisory_json
+      type: string
+      description: |
+          String containing a JSON representation of the advisory data. It should not contain the `advisory`
+          top level key
+          E.g. '{"spec":{"product_id":123,"type":"RHSA"}}'
+    - name: application
+      type: string
+      description: Application being released
+  results:
+    - name: result
+      description: Success if the task succeeds, the error otherwise
+  steps:
+    - name: create-advisory
+      image: quay.io/jbieren/release-utils:ansible_5
+      env:
+        - name: GITLAB_HOST
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_host
+        - name: ACCESS_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: gitlab_access_token
+        - name: GIT_AUTHOR_NAME
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_name
+        - name: GIT_AUTHOR_EMAIL
+          valueFrom:
+            secretKeyRef:
+              name: create-advisory-secret
+              key: git_author_email
+      script: |
+          #!/usr/bin/env sh
+          set -eox pipefail
+
+          REPO_BRANCH=main
+
+          # loading git and gitlab functions
+          . /home/utils/gitlab-functions
+          . /home/utils/git-functions
+          gitlab_init
+          git_functions_init
+
+          function get_advisory_type {
+              NUM_CVES=$(jq '.spec.content.images[].cves.fixed | length' <<< '$(params.advisory_json)' \
+                  | awk '{sum=sum+$0} END{print sum}')
+              if [ $NUM_CVES -gt 0 ] ; then
+                  echo "RHSA"
+                  return
+              fi
+              NUM_ISSUES=$(jq '.spec.issues.fixed | length' <<< '$(params.advisory_json)')
+              if [ $NUM_ISSUES -gt 0 ] ; then
+                  echo "RHBA"
+                  return
+              fi
+              echo "RHEA"
+          }
+
+          function get_advisory_num {
+              # This function will skip many advisory numbers as it is so tied to date. Need to
+              # discover if skipping advisory numbers like this is okay.
+              # There is also a race condition picking the advisory number like this.
+              EXISTING=$(for advisory in $(find data/advisories/*/$(date '+%Y') -name "advisory.yaml") ; do
```

Comment:
There's another instance of `$(date '+%Y')` elsewhere. I think we should only determine the year once and then use that. There's the off chance that the script will run around new year, which is very unlikely, but it would still be cleaner to define the year once.

Reply:
fixed

Reply:
I still see two different calls of `date '+%Y'`, but this code will go away, so I guess it's not a big deal.